### PR TITLE
rdctl: linux/darwin: use process group on factory reset

### PR DESCRIPTION
**Pull request by mook-as**
This adjust rdctl to also use process groups on factory reset, as in #7846.

This will need to land after #7846 and #7838 due to conflicts.

### CODE REVIEW

File: src/go/rdctl/pkg/paths/paths_darwin.go [hunk 1]
```diff
@@ -48,3 +51,38 @@ func GetPaths(getResourcesPathFuncs ...func() (string, error)) (Paths, error) {
...
+		absPathParts := append([]string{dir}, executablePath...)
+		ok, err := checkUsableApplication(filepath.Join(absPathParts...), true)
+		if err != nil {
+			return "", err
+		}
+		if ok {
+			return dir, nil
+		}
+	}
+	return "", errors.New("search locations exhausted")
```
Comment (Nino-K) [inline src/go/rdctl/pkg/paths/paths_darwin.go:73]: Should the returned error be more descriptive to include the locations that were searched (we will thank ourselves during debugging :b)?

File: src/go/rdctl/pkg/paths/paths_linux.go [hunk 2]
```diff
@@ -59,3 +62,38 @@ func GetPaths(getResourcesPathFuncs ...func() (string, error)) (Paths, error) {
...
+	for _, candidatePath := range candidatePaths {
+		usable, err := checkUsableApplication(candidatePath, true)
+		if err != nil {
+			return "", fmt.Errorf("failed to check usability of %q: %w", candidatePath, err)
+		}
+		if usable {
+			return candidatePath, nil
+		}
+	}
+	return "", errors.New("search locations exhausted")
```
Comment (Nino-K) [inline src/go/rdctl/pkg/paths/paths_linux.go:85]: Should this also include the search locations?

File: src/go/rdctl/pkg/shutdown/shutdown.go [hunk 3]
```diff
@@ -201,20 +221,36 @@ func runCommandIgnoreOutput(cmd *exec.Cmd) error {
 	return cmd.Run()
 }
 
-func stopLima() error {
+func stopLima(ctx context.Context) error {
 	return runCommandIgnoreOutput(exec.Command(limaCtlPath, "stop", "0"))
```
Comment (Nino-K) [inline src/go/rdctl/pkg/shutdown/shutdown.go:225]: We are injecting `ctx` we don't even use it, why can't we pass it to `exec.CommandContext(ctx,....`

**Review by Nino-K — COMMENTED** [review]
I just need clarification on including some testing parameters in the actual code to make it testable which is not ideal. I provided a few suggestions about that in some areas, let me know if that makes sense.

File: src/go/rdctl/pkg/directories/directories_test.go [hunk 4]
```diff
@@ -17,15 +17,61 @@ limitations under the License.
...
+		exe, err := os.Create(exePath)
+		require.NoError(t, err)
+		defer exe.Close()
+		link := filepath.Join(dir, "symbolic-link")
+		require.NoError(t, os.Symlink(exePath, link))
+		ctx := directories.OverrideRdctlPath(context.Background(), exePath)
+		actual, err := directories.GetApplicationDirectory(ctx)
+		require.NoError(t, err)
+		assert.Equal(t, dir, actual)
+	})
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/go/rdctl/pkg/directories/directories_test.go:89]: Test cases added. (Obviously the symlink test is skipped on Windows.)

File: src/go/rdctl/pkg/directories/directories_windows.go [hunk 5]
```diff
@@ -83,12 +83,11 @@ func getKnownFolder(folder *windows.KNOWNFOLDERID) (string, error) {
 	)
 	// SHGetKnownFolderPath documentation says we _must_ free the result with
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/go/rdctl/pkg/directories/directories_windows.go:84]: This is not a doc comment, so the link wouldn't work. I can add the URL as plain text though. (The `https://git.io/JMpgD` in the function doc comment actually links to the documentation.)

**Review by Nino-K — APPROVED** [review]

**Pull request merged**